terminus-ui/drawer/src/drawer-container.component.ts

```diff
@@ -0,0 +1,368 @@
+  left: number | null;
+  right: number | null;
+}
+
+const MARGIN_SIZE_CONVERSION = 16;
+
+@Component({
+  selector: 'ts-drawer-container',
+  templateUrl: './drawer-container.component.html',
+  styleUrls: ['./drawer.component.scss'],
+  host: {
+    'class': 'ts-drawer-container',
+    '[class.ts-drawer-container-explicit-backdrop]': '_hasBackdrop',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsDrawerContainer',
+})
+export class TsDrawerContainerComponent implements AfterContentInit, DoCheck, OnDestroy {
+
+  // The drawer at the start/end position.
+  private start: TsDrawerComponent | null = null;
+  private end: TsDrawerComponent | null = null;
+  private openDrawerCount = 0;
+
+  // Emits on every ngDoCheck. Used for debouncing reflows.
+  private readonly doCheckSubject = new Subject<void>();
+
+  // Whether the container is being pushed to the side by one of the drawers.
+  private get isPushed(): boolean {
+    return (TsDrawerContainerComponent.isDrawerOpen(this.start) && this.start.mode === 'push')
+      || (TsDrawerContainerComponent.isDrawerOpen(this.end) && this.end.mode === 'push');
+  }
+
+  public get isShowingBackdrop(): boolean {
+    return (TsDrawerContainerComponent.isDrawerOpen(this.start) && this.hasBackdrop)
+      || (TsDrawerContainerComponent.isDrawerOpen(this.end) && this.hasBackdrop);
+  }
+
+  @ContentChildren(TsDrawerComponent)
+  public drawers!: QueryList<TsDrawerComponent>;
+  @ContentChild(TsDrawerContentComponent, { static: false })
+  public content!: TsDrawerContentComponent;
+
+  /**
+   * Margins to be applied to the content. These are used to push / shrink the drawer content when a
+   * drawer is open. We use margin rather than transform even for push mode because transform breaks
+   * fixed position elements inside of the transformed element.
+   */
+  public contentMargins: ContentLeftRightMargin = {
+    left: null,
+    right: null,
+  };
+
+  public readonly contentMarginChanges = new Subject<ContentLeftRightMargin>();
+
+  /**
+   * Whether the drawer container should have a backdrop while one of the drawers is open.
+   * If explicitly set to `true`, the backdrop will be enabled for drawers.
+   */
+  @Input()
+  public set hasBackdrop(value: boolean) {
+    this._hasBackdrop = value;
+  }
+  public get hasBackdrop() {
+    return this._hasBackdrop;
+  }
+  public _hasBackdrop = false;
+
+  /**
+   * Event emitted when the drawer backdrop is clicked.
+   */
+  @Output()
+  private readonly backdropClick = new EventEmitter<void>();
+
+  constructor(
+    private element: ElementRef<HTMLElement>,
+    private ngZone: NgZone,
+    public renderer: Renderer2,
+    private changeDetectorRef: ChangeDetectorRef,
+    @Optional() @Inject(ANIMATION_MODULE_TYPE) private animationMode?: string
+  ) {
+  }
+
+  public ngAfterContentInit() {
+    const contentEle = this.element.nativeElement.querySelector('.ts-drawer-content');
```

Comment:
Is there a reason we need to do a query here rather than using the view reference created earlier?

Reply:
> The only reason is that content child content isn't always there

This logic should probably be moved into `ngAfterViewInit` then. QueryLists don't exists prior to that lifecycle.

Or I guess lose the querylist and just use this though it doesn't feel as 'nice'